datadog/resource_datadog_logs_metric.go

```diff
@@ -0,0 +1,324 @@
+							Description: "Eventual name of the tag that gets created. By default, the path attribute is used as the tag name.",
+						},
+					},
+				},
+			},
+
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the log-based metric. This field can't be updated after creation.",
+			},
+		},
+	}
+}
+
+func buildDatadogLogsMetric(d *schema.ResourceData) (*datadogV2.LogsMetricCreateData, error) {
+	result := datadogV2.NewLogsMetricCreateDataWithDefaults()
+	result.SetId(d.Get("name").(string))
+
+	attributes := datadogV2.NewLogsMetricCreateAttributesWithDefaults()
+
+	compute, err := getCompute(d)
+	if err != nil {
+		return nil, err
+	}
+	attributes.SetCompute(*compute)
+
+	filter, err := getFilter(d)
+	if err != nil {
+		return nil, err
+	}
+	attributes.SetFilter(*filter)
+
+	groupBys, err := getGroupBys(d)
+	if err != nil {
+		return nil, err
+	}
+	attributes.SetGroupBy(groupBys)
+
+	result.SetAttributes(*attributes)
+	return result, nil
+}
+
+func getCompute(d *schema.ResourceData) (*datadogV2.LogsMetricCompute, error) {
+	resourceCompute := d.Get("compute").([]interface{})[0].(map[string]interface{})
+	compute := datadogV2.NewLogsMetricComputeWithDefaults()
+
+	if aggregationType, ok := resourceCompute["aggregation_type"]; ok {
+		v, err := datadogV2.NewLogsMetricComputeAggregationTypeFromValue(aggregationType.(string))
+		if err != nil {
+			return nil, fmt.Errorf("aggregation_type '%s' is invalid", aggregationType.(string))
+		}
+		compute.SetAggregationType(*v)
+	}
+
+	if path, ok := resourceCompute["path"]; ok {
+		compute.SetPath(path.(string))
+	}
+
+	return compute, nil
+}
+
+func getFilter(d *schema.ResourceData) (*datadogV2.LogsMetricFilter, error) {
+	resourceFilter := d.Get("filter").([]interface{})[0].(map[string]interface{})
+	filter := datadogV2.NewLogsMetricFilterWithDefaults()
+
+	if query, ok := resourceFilter["query"]; ok {
+		filter.SetQuery(query.(string))
+	}
+
+	return filter, nil
+}
+
+func getGroupBys(d *schema.ResourceData) ([]datadogV2.LogsMetricGroupBy, error) {
+	resourceGroupBys := d.Get("group_by").([]interface{})
+	groupBys := make([]datadogV2.LogsMetricGroupBy, len(resourceGroupBys))
+
+	for i, v := range resourceGroupBys {
+		resourceGroupBy := v.(map[string]interface{})
+		groupBy := datadogV2.NewLogsMetricGroupByWithDefaults()
+		if path, ok := resourceGroupBy["path"]; ok {
+			groupBy.SetPath(path.(string))
+		}
+		if path, ok := resourceGroupBy["tag_name"]; ok {
+			groupBy.SetTagName(path.(string))
+		}
+		groupBys[i] = *groupBy
+	}
+
+	return groupBys, nil
+}
+
+func resourceDatadogLogsMetricCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClient := providerConf.DatadogClientV2
+	auth := providerConf.AuthV2
+
+	resultLogsMetricCreateData, err := buildDatadogLogsMetric(d)
+	if err != nil {
+		return translateClientError(err, "error creating LogsMetric")
```

Comment:
nit: I'd do the following to differentiate with the API error below.
```suggestion
		return translateClientError(err, "error building LogsMetric object")
```